sender.go

```diff
@@ -35,6 +35,13 @@ import (
 	"time"
 )
 
+var (
```

Comment:
I'm torn on whether or not to make these errors public. On one had, users can match against them but on the other, it makes the public API messier. wdyt?

Reply:
I'd keep them private for now. Distinguishing them is not very helpful for the user. The only reason I moved them into package-level vars is avoiding allocations.